File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer,Integer> map = new HashMap();

        for(int i=0;i<nums.length; i++){
              int complacent = target - nums[i];
              if(map.containsKey(complacent)){
                  return new int[]{map.get(complacent),i};
              }
            map.put(nums[i],i);
        }

        return new int[]{};
        
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
   let resultObj = {};

    for(let i=0 ; i < nums.length; i++){
        let complement = target - nums[i]
        if(resultObj.hasOwnProperty(complement)){
            return [resultObj[complement] , i];
        }
        resultObj[nums[i]] = i;
    }

    
};